pandas/_libs/tslibs/resolution.pyx

```diff
@@ -39,51 +38,42 @@ def get_resolution(const int64_t[:] stamps, tzinfo tz=None):
         Py_ssize_t i, n = len(stamps)
         npy_datetimestruct dts
         int reso = RESO_DAY, curr_reso
-        ndarray[int64_t] trans
-        int64_t[:] deltas
-        Py_ssize_t[:] pos
-        int64_t local_val, delta
-
-    if is_utc(tz) or tz is None:
-        for i in range(n):
-            if stamps[i] == NPY_NAT:
-                continue
-            dt64_to_dtstruct(stamps[i], &dts)
-            curr_reso = _reso_stamp(&dts)
-            if curr_reso < reso:
-                reso = curr_reso
-    elif is_tzlocal(tz):
+        int64_t local_val
+        TZConvertInfo info
+        ndarray[intp_t, ndim=1] pos2
+
+    info = get_tzconverter(tz, stamps)
+
+    if info.use_fixed:
+        assert info.delta != NPY_NAT
+    elif not info.use_utc and not info.use_tzlocal:
+        assert info.utcoffsets is not NULL
+        assert info.positions is not NULL
+        pos2 = np.array(<intp_t[:n]>info.positions, dtype=np.intp)
         for i in range(n):
-            if stamps[i] == NPY_NAT:
-                continue
+            v1 = info.positions[i]
+            v2 = pos2[i]
+            assert v1 == v2, (v1, v2)
+            assert v1 < info.noffsets, (v1, info.noffsets, i, stamps[i])
```

Comment:
cc @WillAyd I could use a fresh pair of eyes on this.  This debugging assertion is failing on Linux and im at a loss as to why.  The assertion on L265 of tzconversion should behave the same as this, but that one is passing.  Any idea why things could change between there and here?